runtime/dart_isolate.cc

```diff
@@ -709,6 +709,34 @@ static void InvokeDartPluginRegistrantIfAvailable(Dart_Handle library_handle) {
   tonic::LogIfError(tonic::DartInvokeField(plugin_registrant, "register", {}));
 }
 
+namespace {
+bool EndsWith(const std::string& str, const std::string& ending) {
+  if (str.size() >= ending.size()) {
+    return (0 ==
+            str.compare(str.size() - ending.size(), ending.size(), ending));
+  } else {
+    return false;
+  }
+}
+
+Dart_Handle FindDartPluginRegistrantLibrary() {
+  // TODO(): Instead of finding this, it could be passed down from the tool.
+  Dart_Handle libraries = Dart_GetLoadedLibraries();
+  intptr_t length = 0;
+  Dart_ListLength(libraries, &length);
+  for (intptr_t i = 0; i < length; ++i) {
```

Comment:
this grows with the number of packages, right?

Reply:
That's right.  I suspect the order of magnitude will never make this an issue.  It depends on Dart's implementation of their API but linearly searching through a couple thousand strings shouldn't be significant.  Most of the string comparisons will fail on the first char check.

Reply:
Isn't it more like ~275ms?  I might be looking at the wrong thing.

Reply:
Sorry, it was just under 300ms - a mean change of about 291ms on low end Android. This represented a 17% regression from before this patch. I think we should revert this.